packages/nextjs/app/builders/page.tsx

```diff
@@ -0,0 +1,80 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import Image from "next/image";
+import Link from "next/link";
+import BatchRegistryJSON from "../../../hardhat/artifacts/contracts/BatchRegistry.sol/BatchRegistry.json";
+import makeBlockie from "ethereum-blockies-base64";
+import { ethers } from "ethers";
+import type { NextPage } from "next";
+import { GlobeAltIcon } from "@heroicons/react/24/solid";
+
+const BatchRegistryAddress = "0x8C71260fD3eC957faFf9F36Ef9d8C2B988c48E16";
+
+const BuildersPage: NextPage = () => {
+  const [builders, setBuilders] = useState<string[]>([]);
+
+  useEffect(() => {
+    const fetchBuilders = async () => {
+      if (typeof window.ethereum !== "undefined") {
+        const provider = new ethers.providers.Web3Provider(window.ethereum as any);
+        const contract = new ethers.Contract(BatchRegistryAddress, BatchRegistryJSON.abi, provider);
+
+        try {
+          const filter = contract.filters.CheckedIn(null, null, null);
+          const events = await contract.queryFilter(filter, -100000);
```

Comment:
Event's probably a good way to handle it but maybe we should do it the scaffold eth way?

Like the dice game challenge:
```
  const { data: rollsHistoryData, isLoading: rollsHistoryLoading } = useScaffoldEventHistory({
    contractName: "DiceGame",
    eventName: "Roll",
    fromBlock: 0n,
  });
  ```

Reply:
yes! This way you don't have to worry about contract instanciation, keep track of the ABI & address... It's all handled for you.

Reply:
Thanks for the reviewing & insights, @BillSchumacher and @carletex! Submitted a new commit ;)